lib/perl/Genome/InstrumentData/Command/Import/WorkFlow/FastqsToBam.pm

```diff
@@ -57,6 +63,40 @@ sub execute {
     return 1;
 }
 
+sub _unarchive_fastqs_if_necessary {
+    my $self = shift;
+    $self->debug_message('Unarchive fastqs if necessary...');
+
+    my @new_fastq_paths;
+    for my $fastq_path ( $self->fastq_paths ) {
+        if ( $fastq_path !~ /\.gz$/ ) {
```

Comment:
We could probably use [File::Type][1]; it's already packaged, `libfile-type-perl`.

[1]: https://metacpan.org/pod/File::Type